### PR TITLE
Some clean up and improvements

### PR DESCRIPTION
**Pull request by otsch**
* The two types of messages being sent via the `TwitterChannel` class are
  `TwitterStatusUpdate` and `TwitterDirectMessage` and those share a few
  things, so introduce an abstract parent class `TwitterMessage` for them.
* Notifications passed to `TwitterChannel::send()` need to implement a
  method `toTwitter`, so add an Interface `TwitterNotification` defining
  this method and also the return type TwitterMessage. As it'd be
  breaking to change the argument type to this interface, leave this
  open for now.
* Add many argument and return types and remove docblocks.
* Make use of PHP 8 features like constructor property promotion.

### CODE REVIEW

**Comment by christophrumpel** [comment]
This was fast, thanks. Will give it a look. BC are no issues since this will be a new major version of the package 👍

File: README.md [hunk 1]
```diff
@@ -71,10 +71,14 @@ To use this package, you need to create a notification class, like `NewsWasPubli
...
 <?php
+
+use Illuminate\Notifications\Notification;
+use NotificationChannels\Twitter\Contracts\TwitterNotification;
 use NotificationChannels\Twitter\TwitterChannel;
+use NotificationChannels\Twitter\TwitterMessage;
 use NotificationChannels\Twitter\TwitterStatusUpdate;
 
-class NewsWasPublished extends Notification
+class NewsWasPublished extends Notification implements TwitterNotification
```
Comment (christophrumpel) [inline README.md]: I get the idea with the interface for the `toTwitter` method, but I want to keep the example simple. Please remove it.

File: README.md [hunk 2]
```diff
@@ -88,7 +92,11 @@ class NewsWasPublished extends Notification
         return [TwitterChannel::class];
     }
 
-    public function toTwitter($notifiable)
+    /**
+     * @param  mixed $notifiable
+     * @return TwitterMessage
+     */
+    public function toTwitter(mixed $notifiable): TwitterMessage
```
Comment (christophrumpel) [inline README.md:93]: Please remove the docblock where it only adds `params` or `return types` which we already use in the code. In those scenarios, I find it not needed.

File: README.md [hunk 3]
```diff
@@ -97,16 +105,18 @@ class NewsWasPublished extends Notification
 
 Take a closer look at the `toTwitter` method. Here we define what kind of Twitter message we want to trigger. In this case, it is a status update message, which is just a new message in your timeline.
 
+It's recommended to also implement the `TwitterNotification` interface that requires the `toTwitter` method and also that it returns an instance of `TwitterMessage`.
```
Comment (christophrumpel) [inline README.md]: This line is then also not needed anymore.

**Review by christophrumpel — COMMENTED** [review]
Thanks for the great clean-up. 👏  I only had a few notes I added.

File: README.md (hunk 1 above)
Comment (otsch) [reply README.md]: OK, removed 👍🏻

File: README.md (hunk 2 above)
Comment (otsch) [reply README.md:93]: 👍🏻 I've reviewed it again and removed some more redundant docblocks: https://github.com/laravel-notification-channels/twitter/pull/79/commits/b49079a050acf2c3eb13151396a903c87609c5a0

File: README.md (hunk 3 above)
Comment (otsch) [reply README.md]: Removed

**Comment by christophrumpel** [comment]
Thanks a lot, @otsch 👏 There is now another clean-up PR, I will check if there is anything else we can use.

**Pull request merged**